### PR TITLE
docs: expand core-concepts and multi-age

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/evaluating.md [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+The [`method: Page.evaluate`] API can run a JavaScript function in the context
+of the web page and bring results back to the Playwright environment. Browser globals like
+`window` and `document` can be used in `evaluate`.
+
+```js
+const href = await page.evaluate(() => document.location.href);
+```
+
+```java
+String href = (String) page.evaluate("document.location.href");
```
Comment (dgozman) [inline docs/src/evaluating.md:17]: Why string and not a function like other languages?

File: docs/src/evaluating.md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+```python async
+href = await page.evaluate('() => document.location.href')
+```
+
+```python sync
+href = page.evaluate('() => document.location.href')
+```
+
+```csharp
+var href = await page.EvaluateAsync<string>("document.location.href");
```
Comment (dgozman) [inline docs/src/evaluating.md:29]: ditto

File: docs/src/pages.md [hunk 3]
```diff
@@ -1,111 +1,88 @@
 ---
-id: multi-pages
-title: "Multi-page scenarios"
+id: pages
+title: "Pages and frames"
```
Comment (dgozman) [inline docs/src/pages.md:3]: This looks like just "Pages"?

**Review by dgozman — APPROVED** [review]

File: docs/src/evaluating.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/src/evaluating.md:17]: I don't know!

File: docs/src/pages.md (hunk 3 above)
Comment (pavelfeldman) [reply docs/src/pages.md]: Fixed.

**Pull request merged**